### PR TITLE
Implement game logic (get winner)

### PR DESCRIPTION
**Pull request by viettud22**
Find who is the winner - The player that has the most claimed boxes

### CODE REVIEW

File: game.py [hunk 1]
```diff
@@ -67,18 +70,53 @@ def removePlayer(self, player: Player):
...
-    def broadcast(self, message: str):
+    def broadcast(self, message: str):      
         for player in self.players:
-            player.socket.send(message.encode("ascii"))
+            action, *_ = message.split(" ")
+            new_message = message + f" {player.id}"
+            if action == Action.CLAIM:
+                player.socket.send(new_message.encode("ascii"))
+            else:
+                player.socket.send(message.encode("ascii"))
```
Comment (viethung7899) [inline game.py:95]: There's no need to append the playerId end of the message. The playerId is already assigned to the player when they first connects to the game

File: game.py [hunk 2]
```diff
@@ -43,11 +42,39 @@ def isPlaying(self):
...
+        #Get the most repetitive number of claimed boxes
+        c = Counter(claimed_list)
+        highest_score = c.most_common(1) #return [(player index, # of boxes claim)]
+        
+        #If every boxes are filled, we have a winner
+        if(len(claimed_list) == BOARD_SIZE*BOARD_SIZE):
+            winner = highest_score[0]
+            winner_index = winner[0]
+            winner_score = winner[1]
+            print("Winner is player number", winner_index," with highest score: ", winner_score)
```
[the remark this answers is not in the document]
Comment (viettud22) [reply game.py:119]: Sounds good. I added it

File: game.py (hunk 1 above)
Comment (viettud22) [reply game.py]: I removed it. Tks

File: game.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (viethung7899) [reply game.py:114]: It wasn't exactly what we expect, but I think it's okay.

File: game.py [hunk 3]
```diff
@@ -67,13 +70,15 @@ def removePlayer(self, player: Player):
...
+            if len(self.players) < MIN_PLAYERS or len(self.players) > MAX_PLAYERS or self.isGameFinished is True:
                 return
             self.isPlaying = True
             box = self.boxes[row][col]
             if action == Action.HOLD and box.canBeHeld(player):
                 self.broadcast(f"{Action.HOLD} {row} {col} {player.id}")
             elif action == Action.CLAIM and box.canBeClaim(player):
+                if self.checkWinners():
+                    self.isGameFinished = True
                 self.broadcast(f"{Action.CLAIM} {row} {col} {player.id}")
```
Comment (viethung7899) [inline game.py:82]: I think you should broadcast first and then check the winner

**Pull request merged**